packages/smart-wallet/src/offers.js

```diff
@@ -174,6 +179,15 @@ export const makeOfferExecutor = ({
             updateStatus({ result });
           }
         });
+        if (seatRef) {
+          E(seatRef)
+            .hasExited()
+            .then(hasExited => {
+              if (!hasExited) {
+                E(seatRef).tryExit();
+              }
+            });
```

Comment:
MarkM seems to prefer `E.when()` to `.then()`.

```suggestion
          E.when(
            E(seatRef).hasExited(),
            hasExited => {
              if (!hasExited) {
                E(seatRef).tryExit();
              }
            },
          );
```

Reply:
Using `E.when()` always is easier than thinking, but in this case I'm not going to insist: The risk is that the supposed-promise from `hasExited()` would be have a bogus `then` method, but it comes from zoe, and the contract pretty much has to assume zoe is correct; that is: it relies on zoe. In fact, I think this promise comes from `E`, which the contract relies on even more.